deepchem/models/torch_models/grover.py

```diff
@@ -1,94 +1,397 @@
-
-    Parameters
-    ----------
-    edge_fdim: int
-        the dimension of additional feature for edge/bond.
-    node_fdim: int
-        the dimension of additional feature for node/atom.
-    depth: int
-        Dynamic message passing depth for use in MPNEncoder
-    undirected: bool
-        The message passing is undirected or not
-    num_mt_block: int
-        the number of message passing blocks.
-    num_head: int
-        the number of attention heads.
-    atom_embedding_output_type: str
-        the type of output aggregation after message passing.
-                                            atom_messages:      True                      False
-        - "none": no aggregation         output size:     (num_atoms, hidden_size)    (num_bonds, hidden_size)
-        -"atom":  aggregating to atom  output size:     (num_atoms, hidden_size)    (num_atoms, hidden_size)
-        -"bond": aggragating to bond.   output size:     (num_bonds, hidden_size)    (num_bonds, hidden_size)
-        -"both": aggregating to atom&bond. output size:  (num_atoms, hidden_size)    (num_bonds, hidden_size)
-                                                        (num_bonds, hidden_size)    (num_atoms, hidden_size)
-    """
-
-    def __init__(self,
-                 node_fdim,
-                 edge_fdim,
-                 atom_embedding_output_type,
-                 hidden_size=128,
-                 depth=1,
-                 undirected=False,
-                 dropout=0.2,
-                 activation='relu',
-                 num_mt_block=1,
-                 num_heads=4,
-                 bias=False,
-                 res_connection=False):
-        super(GroverEmbedding, self).__init__()
-        self.atom_embedding_output_type = atom_embedding_output_type
-        self.encoders = GroverTransEncoder(
-            hidden_size=hidden_size,
-            edge_fdim=edge_fdim,
-            node_fdim=node_fdim,
-            depth=depth,
-            undirected=undirected,
-            dropout=dropout,
-            activation=activation,
-            num_mt_block=num_mt_block,
-            num_heads=num_heads,
-            atom_emb_output_type=atom_embedding_output_type,
-            bias=bias,
-            res_connection=res_connection)
-
-    def forward(self, graph_batch: List[torch.Tensor]):
-        """Forward function
+import numpy as np
+from typing import List, Dict
+from rdkit import Chem
+from deepchem.models.torch_models.modular import ModularTorchModel
+from deepchem.models.torch_models import TorchModel
+from deepchem.models.torch_models.grover_layers import (
+    GroverEmbedding, GroverBondVocabPredictor, GroverAtomVocabPredictor,
+    GroverFunctionalGroupPredictor)
+from deepchem.models.torch_models.readout import GroverReadout
 
+
+def _grover_batch_mol(mol_graphs):
```

Comment:
This function is really complicated. Can you add a docstring to explain what it does and provide a usage example? I don't usually ask for usage examples on private functions but it will be very hard to debug this function in future otherwise.

You might also consider breaking this into a few simpler functions